sound/soc/sof/intel/hda-dsp.c

```diff
@@ -374,6 +378,12 @@ static int hda_resume(struct snd_sof_dev *sdev)
 	/* enable ppcap interrupt */
 	snd_hdac_ext_bus_ppcap_enable(bus, true);
 	snd_hdac_ext_bus_ppcap_int_enable(bus, true);
+
+	/*
+	 * power cycling of HDA codecs causes failures in HDA
+	 * controller logic -> force codecs to be powered
+	 */
+	snd_hda_set_power_save(hbus, -1);
```

Comment:
this boils down to callling:

	pm_runtime_dont_use_autosuspend(dev);
        pm_runtime_forbid(dev);

which doesn't seem like a very good idea to me, and invalidates what you do on suspend.